resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterManager.scala

```diff
@@ -70,8 +70,17 @@ private[spark] class KubernetesClusterManager extends ExternalClusterManager wit
     // If/when feature steps are executed in client mode, they should instead take care of this,
     // and this code should be removed.
     if (!sc.conf.contains(KUBERNETES_EXECUTOR_POD_NAME_PREFIX)) {
-      sc.conf.set(KUBERNETES_EXECUTOR_POD_NAME_PREFIX,
-        KubernetesConf.getResourceNamePrefix(sc.conf.get("spark.app.name")))
+      val podNamePrefix = KubernetesConf.getResourceNamePrefix(sc.conf.get("spark.app.name"))
+      if (KubernetesUtils.isValidExecutorPodNamePrefix(podNamePrefix)) {
+        sc.conf.set(KUBERNETES_EXECUTOR_POD_NAME_PREFIX, podNamePrefix)
+      } else {
+        val shortPrefix = podNamePrefix.substring(
```

Comment:
The `substring` is not the right method it gives back the suffix:

```
scala> "aaaaaabbbbbb".substring(6)
res1: String = bbbbbb
``` 

It would be more obvious for the user when a prefix of the application name is used

Reply:
thank you @attilapiros , the idea of substring suffix is from `BasicExecutorFeatureStep` in this code:
```
  val hostname = name.substring(Math.max(0, name.length - KUBERNETES_DNSNAME_MAX_LENGTH))
    // Remove non-word characters from the start of the hostname
    .replaceAll("^[^\\w]+", "")
    // Replace dangerous characters in the remaining string with a safe alternative.
    .replaceAll("[^\\w-]+", "_")
```

I'm OK to use the prefix if you have strong opinion, but need to change the code of `KubernetesConf.getResourceNamePrefix`.

Reply:
I don't think it's a big deal one way or the other, but having matching/consistent names would be good.

Reply:
For this one, it seems that we are waiting for @attilapiros 's opinion.

Reply:
I would like to see the prefixes in both cases as that is what a user can identify easily.
So @ulysses-you I would like to kindly ask you to change `BasicExecutorFeatureStep` too!